cluster-state-service/handler/store/taskstore.go

```diff
@@ -154,30 +154,32 @@ func (taskStore eventTaskStore) FilterTasks(filterMap map[string]string) ([]type
 		return nil, errors.Errorf("At least one of the provided filters '%v' is not supported.", filters)
 	}
 
-	status := filterMap[taskStatusFilter]
-	cluster := filterMap[taskClusterFilter]
-	startedBy := filterMap[taskStartedByFilter]
-
-	// Currently the only filter combination that is supported is status & cluster
-	if len(filters) == 2 {
-		if status != "" && cluster != "" {
-			return taskStore.filterTasksByClusterAndStatus(cluster, status)
+	var result []types.Task
+	var err error
+	if cluster := filterMap[taskClusterFilter]; cluster != "" {
+		result, err = taskStore.filterTasksByCluster(cluster)
```

Comment:
May be worthwhile to add a comment that says filter by cluster does a etcd list by cluster prefix.